dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/TeiAnalyticsQueryService.java

```diff
@@ -72,6 +74,10 @@
      */
     public Grid getGrid( final TeiQueryParams teiQueryParams, final CommonQueryRequest commonQueryRequest )
     {
+        Query queryNg = QueryContext.of( teiQueryParams ).getQuery();
+        // output should look like: https://pastebin.com/4aK9ZxEQ
```

Comment:
Nice to have some unit tests to assert the expected query, using different parameters/scenarios.
(I can assist you with unit test coverage @gnespolino)